Private events project features code review.

dipbazz opened the pull request:
In this project we created a private events web application where;

- A user can create events.
- A user can attend many events
- An event can be attended by many users.

The models used for this project are:

- `User` model with field `username`
-  `Event` model with field `date`, `description`
-  `Attendence` model for association between `User` as `attendee` and `Event` as `attended_event`

The features of this project in this pull request are:

- On the home page all lists of events will be displayed as past (previous events) and future (upcoming events).
- User can log in to the app.
- User can logout from the app.
- User can view their profile with the list of events they have joined in the past (previous events) and the events they are joined to attend in the future (upcoming events).
- Logged in users can join the event if they haven't joined yet.
- User can create an event.

dipbazz committed:
Fix rubocop errors and rename file name

Byabasaija committed:
add user model

add  user controller methods
add simple user sign up form

dipbazz committed:
Add global methods to check for user session

Byabasaija committed:
add association btn creator and event

modify show page to show events created by user

dipbazz committed:
Add event controller and event description migration

Add description to event model and in event controller methods
for displaying all events #index, and creating the event create with
respctive view.

Byabasaija committed:
Add attendence associations

dipbazz committed:
Update show view and show method

Show view to display a list attendees attending the event
and also show view in user to display the list of event
the user is going to attend

Byabasaija committed:
Modify users show page and Events show page to;

Upcoming events from previous ones
Apply scopes to past and upcoming methods

dipbazz committed:
Update style of the project and fix some bug

add bootstrap gem file and update the index page of events
to show the events in card layout for upcomming and previous
events. Add navabar and alert for flash message in
application.html.erb file.

Byabasaija committed:
Complete styling

Byabasaija committed:
Complete the project styling

dipbazz committed:
Add event join feature and fix rubocop errors

oloomoses commented on app/views/events/show.html.erb:
```diff
@@ -0,0 +1,27 @@
+              <hr class="my-4">
+              <h6>Created by:</h6>
+              <p><%= @event.creator.username %></p>
+              <% if @event.attendees.exists? %>
+                <h3>List of Attendees</h3>
+              <ul>
+                <% @event.attendees.each do |attendee| %>
+                  <li><%= attendee.username %></li>
+                <% end %>
+              </ul>
+              <% end %>
```
Remember to your logic into the helpers

oloomoses reviewed (changes requested):
# Status: Changes Required
Hello @dipbazz and @Byabasaija, You are doing a very good job on this project :muscle: 
The structure of your pull request is good and very easy to understand :clap: 
Some changes to make before the project can be approved :point_down: 
## Required
Ensure that:
- [x] a logged-in user can attend an event
- [x] a logged-in user is able to logout 
- [x] the user's show page displays events the user has created and events that the user is attending to separated in past and upcoming
- [x] the event's show page displays the list of attendants
##### EventsController
###### `create` action

* [x] uses #build method on association, i.e. `current_user.events.build(event_params)` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/build_method_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

###### `index` action
* [x] lists past events, i.e. `@past_events = Event.previous_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [x] lists ucomping events, i.e. `@upcoming_events = Event.upcoming_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

##### UsersController
###### `show` action
* [x] lists events created by user, i.e. `@created_events = current_user.events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/current_user_show_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [x] lists events previoulsy attendented by user **using event model scope**, i.e. `@past_events = current_user.attended_events.previous` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [x] lists upcoming events previously attended by user **using event model scope**, i.e. `@upcoming_events = current_user.attended_events.upcoming` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

## Feel free to reach me on slack [ Oloo Moses ] incase of doubts or concerns
# Oloo Moses

dipbazz committed:
Update eventscontroller with required changes

create action uses #build method for current user to create the
events.

Index action has the instance variable regarding the past events
and upcoming events not in the view.

Byabasaija committed:
Modify show action in users controller

update the show pages

MahmoudBakr23 reviewed (changes requested):
# **Changes Required ♻🛑**

<hr>

Hello there, great job you've done on this project and on the requested changes by the previous Code Reviewers. However, there is still one change you need to make in order to get this pull request approved!

<hr>

- [x] Please fix your Login/Logout logic and functionalities. Try to refactor your sessions logic by creating a specific controller for the sessions (If you see appropriate, you can fix the current logic to make it work properly)

<hr>

Kindly fix the mandatory change above and ask for another re-review 🙏😊

Good luck and happy coding 👨‍💻💻

Byabasaija committed:
fix the log in log out

mimipeshy commented:
### STATUS :APPROVED :partying_face: :champagne: 

Hey team @Byabasaija  and @dipbazz, you guys have done an amazing job with the feedback given.  This project is therefore approved. Go ahead and merge

All the best in the next projects. Happy Coding :computer: 

Regards
Peris Ndanu
